common/app/model/content/Atoms.scala

```diff
@@ -53,6 +54,11 @@ final case class MediaAtom(
       "^0".r.replaceFirstIn(formattedDuration, "") //strip leading zero
     }
   }
+
+  def activeAsset: MediaAsset = activeVersion match {
+    case Some(version) => assets.find(_.version == version).getOrElse(assets.head)
+    case _ => assets.head
+  }
```

Comment:
I've made it an Option and moved it to YouTube specific helper method. @akash1810 hopefully that makes it as obvious as a comment would?